app/jinja_filters.py

```diff
@@ -311,6 +310,15 @@ def format_date_range_no_repeated_month_year(context, start_date, end_date, date
     return mark_safe(context, output)
 
 
+@evalcontextfunction
+def one_of(context, *args):
```

Comment:
Could you also refactor `format_conditional_date` to reduce duplication?

Reply:
Make some changes.